rollup/sync_service.go

```diff
@@ -644,326 +238,126 @@ func (s *SyncService) Stop() error {
-		FromBlock: header.Number,
-		ToBlock:   header.Number,
-		Topics:    [][]common.Hash{},
+	// The transaction does not yet exist in the ctc
+	if tx == nil {
+		log.Trace("Transaction in ctc does not yet exist", "index", index)
+		return nil
 	}
 
-	logs, err := s.logClient.FilterLogs(ctx, query)
+	err = s.applyTransaction(tx)
 	if err != nil {
-		return s.Eth1Data, fmt.Errorf("Cannot query for logs at block %s: %w", blockHash.Hex(), err)
-	}
-
-	// sort the logs by Index
-	// TODO: test ByIndex, must be ascending
-	sort.Sort(LogsByIndex(logs))
-	for _, ethlog := range logs {
-		if ethlog.BlockNumber != blockHeight {
-			log.Warn("Unexpected block height from log", "got", ethlog.BlockNumber, "expected", blockHeight)
-			continue
-		}
-		// Prevent logs emitted from other contracts from being processed
-		if !bytes.Equal(ethlog.Address.Bytes(), s.CanonicalTransactionChainAddress.Bytes()) {
-			continue
-		}
-		if err := s.ProcessLog(ctx, ethlog); err != nil {
-			// TODO: reorg out the applied transactions and remove the
-			// transactions that were added to the cache so that none are
-			// replayed. The same Eth1Data is returned, so it will not be
-			// updated. In `processHistoricalLogs`, this will result in the same
-			// block being queried. In the `Loop`, the next block should arrive
-			// via a notification. Think about good solutions for this.
-			return s.Eth1Data, fmt.Errorf("Cannot process log at height %d: %w", blockHeight, err)
-		}
+		return fmt.Errorf("Cannot apply transaction: %w", err)
 	}
 
-	// Write to the database for term persistence
-	rawdb.WriteHeadEth1HeaderHash(s.db, header.Hash())
-	rawdb.WriteHeadEth1HeaderHeight(s.db, blockHeight)
-	s.HeaderCache[blockHeight%headerCacheSize] = header
-
-	return Eth1Data{
-		BlockHash:   blockHash,
-		BlockHeight: blockHeight,
-	}, nil
+	return nil
 }
 
-// ApplyLogs will apply cached transactions from `enqueue` logs.
-// This function should only be called in the case of sequencer.
-func (s *SyncService) ApplyLogs(tip *types.Header) error {
-	// Handle the enqueue'd transactions. This codepath is only useful
-	// for the sequencer, as the verifier should only handle transactions
-	// from sequencer batch append and queue batch append.
-	// The transactions need to be played in order and there is no
-	// guarantee of order when it comes to the txcache iteration, so
-	// collect an array of pointers and then sort them by index.
-	txs := []*RollupTransaction{}
-	s.txCache.Range(func(index uint64, rtx *RollupTransaction) {
-		// The transaction has not been executed. We know that it is
-		// sufficiently old because we only ingest blocks that are
-		// sufficiently old.
-		if !rtx.executed {
-			txs = append(txs, rtx)
-		}
-	})
+func (s *SyncService) syncTransactionsToTip() error {
+	latest, err := s.client.GetLatestTransaction()
+	if err != nil {
+		return fmt.Errorf("Cannot get latest transaction: %w", err)
+	}
+	block := s.bc.CurrentBlock()
+	tipHeight := latest.GetMeta().Index
```

Comment:
Double check an off by one error here because the index and block numbers are offset by 1